app/agents/research.py

```
@@ -0,0 +1,401 @@
+        if key in product_type.lower():
+            candidates = mock_data[key]
+            break
+
+    # Fallback to generic product if no match
+    if candidates is None:
+        logger.warning(f"No mock data for product type '{product_type}', using generic fallback")
+        candidates = [
+            {
+                "name": f"Generic {product_type.title()} Option 1",
+                "manufacturer": "Unknown",
+                "official_url": f"https://example.com/product-1",
```

Ruff (0.14.14)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)